src/helper/convertColumnElementsToData.js

```diff
@@ -99,4 +112,85 @@ function convertColumnElementsToData(childComponents) {
   };
 };
 
-export default convertColumnElementsToData;
+/**
+ *
+ */
+function collectPropsAndTemplatesFromColumnData(column, columnProps, elementTemplates) {
+  columnProps.push(_extractPropsFromData(column));
+  _extractTemplatesFromData(elementTemplates, column);
+}
+
+/**
+ * Uses columnGetter to fetch column props and cell rendering templates
+ *
+ * @param {!function(number)} columnGetter
+ * @param {!number} columnsCount
+ * @return {{
+ *   columnGroupProps: !Array.<columnDefinition>,
+ *   columnProps: !Array.<columnDefinition>,
+ *   elementTemplates: !Object.<string, Array>,
+ *   useGroupHeader: boolean,
+ * }}
+ */
+function fetchColumnData(columnGetter, columnsCount) {
+  let columnProps = [];
+  let columnGroupProps = [];
+  let columnsWithDefinedGroupIndex = 0;
+  let columnGroups = [];
+
+  const elementTemplates = getDefaultElementTemplates();
+
+  // fetch column props and templates for each column
+  for (let index = 0; index < columnsCount; index++) {
+    collectPropsAndTemplatesFromColumnData(columnGetter({ index }), columnProps, elementTemplates);
+  }
+
+  // keep a map of group indexes which will be used to fetch the column groups
+  forEach(columnProps, (column) => {
+    const groupIndex = column.groupIdx;
+    if (groupIndex !== undefined && groupIndex !== null) {
+      columnGroups[groupIndex] = true;
+      columnsWithDefinedGroupIndex++;
+    }
+  });
+
+  // if group index was specified, it should be given for every column
+  if (columnsWithDefinedGroupIndex > 0) {
```

Comment:
just a useful check since FDT currently doesn't support mixing ungrouped and grouped columns

(probably #305 #174)